paper-server/src/main/java/io/papermc/paper/raytracing/PositionedRayTraceConfigurationBuilderImpl.java

```diff
@@ -0,0 +1,129 @@
+    private double maxDistance;
+    private org.bukkit.FluidCollisionMode fluidCollisionMode = org.bukkit.FluidCollisionMode.NEVER;
+    private boolean ignorePassableBlocks;
+    private double raySize = 0.0D;
+    private java.util.function.Predicate<? super org.bukkit.entity.Entity> entityFilter;
+    private java.util.function.Predicate<? super org.bukkit.block.Block> blockFilter;
+    private List<RayTraceTarget> targets;
+
+    @Override
+    public Location start() {
+        return this.start == null ? null : this.start.clone();
+    }
+
+    @Override
+    public PositionedRayTraceConfigurationBuilder start(final Location start) {
+        this.start = start.clone();
+        return this;
+    }
+
+    @Override
+    public Vector direction() {
+        return this.direction == null ? null : this.direction.clone();
+    }
+
+    @Override
+    public PositionedRayTraceConfigurationBuilder direction(final Vector direction) {
+        this.direction = direction.clone();
+        return this;
+    }
+
+    @Override
+    public double maxDistance() {
+        return maxDistance;
+    }
+
+    @Override
+    public PositionedRayTraceConfigurationBuilder maxDistance(double maxDistance) {
+        this.maxDistance = maxDistance;
+        return this;
+    }
+
+    @Override
+    public FluidCollisionMode fluidCollisionMode() {
+        return fluidCollisionMode;
+    }
+
+    @Override
+    public PositionedRayTraceConfigurationBuilder fluidCollisionMode(FluidCollisionMode fluidCollisionMode) {
+        this.fluidCollisionMode = fluidCollisionMode;
+        return this;
+    }
+
+    @Override
+    public boolean ignorePassableBlocks() {
+        return ignorePassableBlocks;
+    }
+
+    @Override
+    public PositionedRayTraceConfigurationBuilder ignorePassableBlocks(boolean ignorePassableBlocks) {
+        this.ignorePassableBlocks = ignorePassableBlocks;
+        return this;
+    }
+
+    @Override
+    public double raySize() {
+        return raySize;
+    }
+
+    @Override
+    public PositionedRayTraceConfigurationBuilder raySize(double raySize) {
+        this.raySize = raySize;
+        return this;
+    }
+
+    @Override
+    public Predicate<? super org.bukkit.entity.Entity> entityFilter() {
+        return entityFilter;
+    }
+
+    @Override
+    public PositionedRayTraceConfigurationBuilder entityFilter(Predicate<? super org.bukkit.entity.Entity> entityFilter) {
+        this.entityFilter = entityFilter;
+        return this;
+    }
+
+    @Override
+    public Predicate<? super org.bukkit.block.Block> blockFilter() {
+        return blockFilter;
+    }
+
+    @Override
+    public PositionedRayTraceConfigurationBuilder blockFilter(Predicate<? super org.bukkit.block.Block> blockFilter) {
+        this.blockFilter = blockFilter;
+        return this;
+    }
+
+    @Override
+    public @UnmodifiableView List<RayTraceTarget> targets() {
+        return Collections.unmodifiableList(this.targets);
```

Comment:
For builder without targets this will cause a NPE, might be better to make it a requirement